presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -689,9 +704,11 @@ protected Node visitSimpleGroupBy(SimpleGroupBy node, Void context)
 
         private boolean validateExpressionForGroupBy(Set<Expression> groupByOfMaterializedView, Expression expression)
         {
+            boolean canRewrite = expression instanceof FunctionCall && SUPPORTED_FUNCTION_CALLS.contains(((FunctionCall) expression).getName());
```

Comment:
This implementation has got bit hacky. This method by itself is not correctly validating if this rewrite operation would be successful.
It is true that rewrite operation itself would correctly determine if the query can be rewritten or not.
I am okay with not fixing this as part of this PR, but lets create a detailed github issue and link it here.

Reply:
Added a validate to `MaterializedViewUtils` to handle this.